src/lib/isJSON.js

```diff
@@ -1,6 +1,9 @@
 import assertString from './util/assertString';
 
 export default function isJSON(str) {
+  if (typeof str === 'object') {
```

Comment:
why don't you wait till the `JSON.parse(str)` stage?

Reply:
Because assertString stoping me to get there, it checks if str is a string or not. In my case, it is an object, so it never pass the assertString to get to the JSON.parse(str)

Reply:
What if we try to change the data send to `string` before using `assertString()`? 
I am saying this because JSON object are much like javascript objects but not the same thing.

Reply:
Also, if we do as I proposed, lint might not allow us to write its test, it will return an error?